### PR TITLE
BDRSPS-672 Add documentation for the template schema.json schema

### PR DESCRIPTION
**Pull request by joecrowleygaia**
- Adds a programmatic way of generating markdown for pydantic models
- This PR focuses on the `schema.json` model only

### CODE REVIEW

File: abis_mapping/types/schema.py [hunk 1]
```diff
@@ -7,31 +7,95 @@
...
+    ]
+    url: Annotated[pydantic.AnyUrl | None, pydantic.Field(description="The IRI of the field's concept.")] = None
     constraints: Constraints
-    vocabularies: list[str] = []
+    vocabularies: Annotated[
+        list[str],
+        pydantic.Field(
+            description="Optional list of vocabulary IDs, defined internally within the project. Provided IDs need to have been registered to be valid. See [`abis_mapping.vocabs`](/abis_mapping/vocabs/)."
+        ),
+    ] = []
```
Comment (Lincoln-GR) [inline abis_mapping/types/schema.py:98]: improvement - swap `= []` for `default_factory=list` to avoid having a shared mutable default

**Review by Lincoln-GR — COMMENTED** [review]
Looks good, few small comments

File: README.md [hunk 2]
```diff
@@ -7,18 +7,40 @@ representation conforming to the Australia Biodiversity Information Standard (AB
...
+## Templates
+
+A template is at its core a CSV document that is used to take tabular data from an outside source and transform
+it into ABIS conformant RDF.
+
+A template is defined through a set of key files
+
+* A CSV file containing a single header row, each cell containing a field name (**required**).
+* A `schema.json` file, containing schema definitions for each of the fields declared in the CSV (**required**).
+* A `metadata.json` file, containing overall descriptions and other key values (**required**). See the detailed definition [here](/docs/models/markdown/Schema.schema.md).
```
[the remark this answers is not in the document]
Comment (joecrowleygaia) [reply README.md]: fixed

File: tests/docs/test_model_json.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    Args:
+        mocker: The mocker fixture.
+    """
+    # Patch open
+    mocked_open = mocker.mock_open()
+    print(__name__)
+    mocker.patch("docs.model_json.open", mocked_open)
+
+    # Invoke
+    try:
```
[the remark this answers is not in the document]
Comment (joecrowleygaia) [reply tests/docs/test_model_json.py]: nope. removed.

File: abis_mapping/types/schema.py (hunk 1 above)
Comment (joecrowleygaia) [reply abis_mapping/types/schema.py]: implemented

File: tests/docs/test_model_json.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+def test_create_json_schema(mocker: pytest_mock.MockerFixture) -> None:
+    """Tests the create_json_schema function.
+
+    Args:
+        mocker: The mocker fixture.
+    """
+    # Patch open
+    mocked_open = mocker.mock_open()
+    print(__name__)
```
[the remark this answers is not in the document]
Comment (joecrowleygaia) [reply tests/docs/test_model_json.py]: removed.

**Review by Lincoln-GR — APPROVED** [review]

**Comment by serge-gaia** [comment]
Great job @joecrowleygaia!

**Pull request merged**